engine/table/src/test/java/io/deephaven/engine/table/impl/TestRollup.java

```diff
@@ -132,6 +135,56 @@ private void testRollupIncrementalInternal(final String ctxt, final int size) {
         }
     }
 
+    @Test
+    public void testRollupWithFilter() {
+        final Table source = newTable(
+                stringCol("GRP", "v1", "v1", "v2", "v2", "v3"),
+                intCol("AVG", 1, 2, 3, 4, 5),
+                intCol("CONST", 11, 12, 13, 14, 15));
+        final List<Aggregation> aggs = List.of(AggAvg("avg=AVG"));
```

Comment:
We could actually validate that our intention of not recreating the entire rollup on the grouping only filters is functional, with something like this:
```suggestion
        final Table sourceUncounted = newTable(
                stringCol("GRP", "v1", "v1", "v2", "v2", "v3"),
                intCol("AVG", 1, 2, 3, 4, 5),
                intCol("CONST", 11, 12, 13, 14, 15));

        final AtomicInteger grpCount = new AtomicInteger(0);
        final Function<String, String> countingIdentity = (s) -> { grpCount.incrementAndGet(); return s; };
        QueryScope.addParam("countingIdentity", countingIdentity);
        final Table source = sourceUncounted.updateView("GRP=(String)countingIdentity.apply(GRP)");
```